src/pages/Room/CalendarContainer.jsx

```diff
@@ -1,91 +1,82 @@
-      const allData = await bookingService.getAllWithDetails(
-        getUTCDateFilter()
-      );
-      const data = await bookingService.getAllWithDetailsByRoom(
-        getUTCDateFilter(),
-        URLRoomId
-      );
-      if (data.bookings && allData.bookings) {
-        const { bookings: allBookingsData } = allData;
-        const { bookings: bookingsData } = data;
-        this.setState({ bookingsData, allBookingsData, isServerDown: false });
-      } else {
-        const { name } = data;
-        // The webtoken is invalid for any reason
-        if (name === "JsonWebTokenError") {
-          onLogout();
-          return history.push("/login");
+  // fetchBookings can't be called as `useEffect` first param because it's async
+  useEffect(() => {
+    const fetchBookings = async () => {
+      try {
+        const { onLogout } = authContext;
+        const reqRoom = await roomService.getOneById(URLRoomId);
+        const allData = await bookingService.getAllWithDetails(
+          getUTCDateFilter()
+        );
+        const data = await bookingService.getAllWithDetailsByRoom(
+          getUTCDateFilter(),
+          URLRoomId
+        );
+        if (data.bookings && allData.bookings) {
+          const { bookings: allBookingsData } = allData;
+          const { bookings: bookingsData } = data;
+          setAllBookingsData(allBookingsData);
+          setBookingsData(bookingsData);
         } else {
-          this.setState({ isServerDown: true });
+          const { name } = data;
+          // The webtoken is invalid for any reason
+          if (name === "JsonWebTokenError") {
+            onLogout();
+            history.push("/login");
+            return undefined;
+          } else {
+            setIsServerDown(true);
+          }
         }
+        if (typeof reqRoom === "object") {
+          setIsLoading(false);
+          setRoomId(URLRoomId);
+          return undefined;
+        }
+        setIsLoading(true);
+        return undefined;
+      } catch (error) {
+        // Don't know how to do this with hooks
+        Promise.reject(new Error(error.message));
+        return undefined;
       }
+    };
 
-      if (typeof reqRoom === "object") {
-        return this.setState({ roomId: URLRoomId, isLoading: false });
-      }
-      return this.setState({ roomId: "", isLoading: false });
-    } catch (error) {
-      return Promise.reject(new Error(error.message));
-    }
-  };
-
-  handleBookingsDataChange = () => {
-    this.fetchBookings();
-  };
+    fetchBookings();
+    // whenever shouldFetch changes, it will call `fetchBookings`
+  }, [shouldFetch]);
 
-  render() {
-    const {
-      allBookingsData,
-      bookingsData,
-      isServerDown,
-      roomId,
-      isLoading
-    } = this.state;
+  if (isServerDown) {
+    return <Error500 />;
+  }
 
-    if (isServerDown) {
-      return <Error500 />;
-    }
+  const onBookingsDataChange = () => setShouldFetch(true);
```

Comment:
once it is setted to true there's no way to fetch again, do you think is there any other way to do this?

Reply:
Sure, I missed that one.